internal/external-dns/plan/plan.go

```diff
@@ -200,6 +206,60 @@ func (p *Plan) Error() error {
 	return errors.Join(p.Errors...)
 }
 
+func CleanLabelFromEndpoint(label string, endpoint *endpoint.Endpoint) *endpoint.Endpoint {
+	if endpoint.Labels == nil {
+		return endpoint
+	}
+
+	delete(endpoint.Labels, label)
+	return endpoint
+}
+
+func CleanLabelFromEndpoints(label string, endpoints []*endpoint.Endpoint) []*endpoint.Endpoint {
+	for _, e := range endpoints {
+		CleanLabelFromEndpoint(label, e)
+	}
+	return endpoints
+}
+func CleanLabelsFromEndpoints(labels []string, endpoints []*endpoint.Endpoint) []*endpoint.Endpoint {
+	for _, l := range labels {
+		CleanLabelFromEndpoints(l, endpoints)
+	}
+
+	return endpoints
+}
```

Comment:
nit: Might just be me, but  "Remove" or "Delete" seem more appropriate than "Clean" i.e RemoveLabelsFromEndpoints. These are also really helpers nothing to do with the plan, maybe they would be better elsewhere?